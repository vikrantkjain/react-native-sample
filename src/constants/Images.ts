/**
 * Declare all the image name and path here for using in throughout the app.
 */
export const Images = {
  ic_check: require('../assets/images/ic_check.png'),
  ic_drop_down: require('../assets/images/ic_drop_down.png'),
  ic_eye_visible: require('../assets/images/ic_eye_visible.png'),
  ic_eye_hidden: require('../assets/images/ic_eye_hidden.png'),
  ic_green_star: require('../assets/images/ic_green_star.png'),
  ic_grey_star: require('../assets/images/ic_grey_star.png'),
  ic_hal_star_rating_yellow: require('../assets/images/ic_hal_star_rating_yellow.png'),
  ic_half_star_green_rating: require('../assets/images/ic_half_star_green_rating.png'),
  ic_yellow_star: require('../assets/images/ic_yellow_star.png'),
  ic_uncheck: require('../assets/images/ic_uncheck.png'),
  ic_left: require('../assets/images/ic_left.png'),
  ic_star: require('../assets/images/ic_star.png'),
  ic_minus: require('../assets/images/ic_minus.png'),
  ic_plus: require('../assets/images/ic_plus.png'),
  ic_drawer: require('../assets/icons/ic_drawer.png'),
  ic_user: require('../assets/icons/ic_user.png'),
  ic_search: require('../assets/icons/ic_search.png'),
  ic_home: require('../assets/icons/ic_home.png'),
  ic_history: require('../assets/icons/ic_history.png'),
  ic_menu: require('../assets/icons/ic_menu.png'),
  ic_back: require('../assets/icons/ic_back.png'),
  ic_back_black: require('../assets/icons/ic_back_black.png'),
  ic_rotate_Camera: require('../assets/images/rotateCamera.png'),
  ic_flash: require('../assets/images/flash.png'),
  ic_cross: require('../assets/images/ic_minus.png'),
  ic_demoCodeLogo: require('../assets/icons/ic_demoCodeLogo.png'),
  ic_google: require('../assets/icons/ic_google.png'),
  ic_logout: require('../assets/icons/ic_logout.png'),
};

export default Images;
