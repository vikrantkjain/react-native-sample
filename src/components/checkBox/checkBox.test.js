import React from 'react';
import renderer from 'react-test-renderer';
import CheckBox from './index';
import Images from '../../constants/Images';
test('CheckBox render correctly', () => {
  const tree = renderer
    .create(
      <CheckBox
        checkedImage={Images.ic_check}
        uncheckedImage={Images.ic_uncheck}
        checkedImageStyle={{
          backgroundColor: 'yellow',
        }}
        uncheckedImageStyle={{
          backgroundColor: 'red',
        }}
        style={{height: 10, width: 10}}>
        Hey
      </CheckBox>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
