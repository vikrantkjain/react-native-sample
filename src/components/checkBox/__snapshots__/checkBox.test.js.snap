// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 10,
      "opacity": 1,
      "width": 10,
    }
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        [
          [
            {
              "height": 20,
              "width": 20,
            },
            {
              "backgroundColor": "red",
            },
          ],
        ],
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="contain"
      source={
        {
          "testUri": "../../../src/assets/images/ic_uncheck.png",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  Hey
</View>
`;
